lib/index.js

```diff
@@ -0,0 +1,143 @@
+async function goHome() {
+  await browser.goTo('#Shell-home');
+  await browser.waitForUI5();
+}
+
+async function pressTile(name) {
+  const control = await browser.asControl({
+    selector: {
+      controlType: "sap.m.GenericTile",
+      properties: {
+        header: name
+      }
+    }
+  });
+  await control.press()
+}
+
+async function searchFor(text) {
+  await browser.asControl({
+    selector: {
+      controlType: "sap.m.SearchField",
+      interaction: {
+        idSuffix: "I"
+      }
+    }
+  }).enterText(text)
+}
+
+async function performStandardAction( name ) {  
+  const control = await browser.asControl({
+    selector: {
+      controlType: "sap.m.Button",
+      interactable: true,
+      id: new RegExp(`.*\:\:StandardAction\:\:${name}\$`)
+    }
+  });
+  await control.press();
+}
+
+async function setFieldValue( name, value ) {
+  const control = await browser.asControl({
+    selector: {
+      controlType: "sap.m.Input",
+      interaction: "root",
+      labelFor: {
+        text: name
+      }
+    }
+  });
+  await control.setValue(value);
+}
+
+async function openValueHelpForField( name ) {
+  const control = await browser.asControl({
+    selector: {
+      controlType: "sap.m.Input",
+      labelFor: {
+        text: name
+      }
+    }
+  });
+  await control.press();
+}
+
+async function _findTableWithTitle(title, controlType, opt={}) {
+  let { searchOpenDialogs } = opt;
+  return await browser.asControl({
+    selector: {
+      controlType,
+      searchOpenDialogs,
+      interaction: "root",
+      descendant: {
+        controlType: "sap.m.Title",
+        properties: {
+          text: {
+            regex: {
+              source: `^${title}`
+            }
+          }
+        }
+      }
+    }
+  })
+}
+
+async function chooseRowInValueHelpDialogTable( tableTitle, rowNumber ) {
+  let table = await _findTableWithTitle(tableTitle, "sap.ui.table.Table", {searchOpenDialogs:true});
+  //let rows = await table.getAggregation('rows');
+  //let cells = await rows[rowNumber].getAggregation('cells');
+  //let id = await cells[0].getId();
+  let id = await table.getId()+`-rows-row${rowNumber}-col0`;
+  await tapOnId(id);
+}
+
+async function selectRowInTable( tableTitle, targetRow ) {
+  let table = await _findTableWithTitle(tableTitle, "sap.m.Table");
+  let items = await table.getItems();
```

Comment:
this has proven to be a potential drawback: if the table in question has `growing` turned off and at the same time many entries, the test will get flaky here with looong runtimes. Don't have an immediate suggestion for mitigation, just be aware and eventually leave a comment here à la `TODO` or `REVISIT`

Reply:
added a comment